src/transport/mqtt/mod.rs

```diff
@@ -787,26 +780,24 @@ impl SessionData {
     }
 }
 
-#[async_trait]
 trait AsyncClientExt {
     /// Sends the introspection [`String`].
-    async fn send_introspection(
+    fn send_introspection(
```

Comment:
Is it not possible to keep sugar version `async`, since it is a private trait?
```suggestion
    async fn send_introspection(
    ...
    ) -> Result<NoticeFuture, ClientError>;
```